### PR TITLE
chore (core): rename model registry to provider registry

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
- rename `ModelRegistry` to `ProviderRegistry`

## Follow up work
- setup redirects
- add embedding model support to registry

### CODE REVIEW

File: content/docs/07-reference/ai-sdk-core/40-provider-registry.mdx [hunk 1]
```diff
@@ -1,28 +1,28 @@
 ---
-title: experimental_createModelRegistry
+title: experimental_createProviderRegistry
 description: Registry for managing multiple providers and models (API Reference)
 ---
 
-# `experimental_createModelRegistry()`
+# `experimental_createProviderRegistry()`
 
 <Note>Model management is an experimental feature.</Note>
```
Comment (iteratetograceness) [inline content/docs/07-reference/ai-sdk-core/40-provider-registry.mdx:8]:
```suggestion
<Note>Provider management is an experimental feature.</Note>
```

File: content/docs/07-reference/ai-sdk-core/40-provider-registry.mdx [hunk 2]
```diff
@@ -72,7 +72,7 @@ Registers a language model provider with a given id.
 
 ### Returns
 
-The `experimental_createModelRegistry` function returns a `experimental_ModelRegistry` instance. It has the following methods:
+The `experimental_createProviderRegistry` function returns a `experimental_ModelRegistry` instance. It has the following methods:
```
Comment (iteratetograceness) [inline content/docs/07-reference/ai-sdk-core/40-provider-registry.mdx:75]:
```suggestion
The `experimental_createProviderRegistry` function returns a `experimental_ProviderRegistry` instance. It has the following methods:
```

assuming this should be `experimental_ProviderRegistry` instance!

Comment (iteratetograceness) [reply content/docs/07-reference/ai-sdk-core/40-provider-registry.mdx:75]: pls ignore if not

**Review by iteratetograceness — APPROVED** [review]

File: packages/core/core/registry/provider-registry.ts [hunk 3]
```diff
@@ -36,7 +41,7 @@ export function experimental_createModelRegistry(
   return registry;
 }
 
-class DefaultModelRegistry implements experimental_ModelRegistry {
+class DefaultModelRegistry implements experimental_ProviderRegistry {
```
Comment (lgrammel) [inline packages/core/core/registry/provider-registry.ts:44]: hm i also need to rename here

**Pull request merged**